src/main/java/cms/gov/madie/measure/utils/GroupPopulationUtil.java

```diff
@@ -56,4 +51,64 @@ static String getPopulationDisplayId(
     }
     return newPopDisplayId;
   }
+
+  static void setPopulationDisplayIds(Group group, String groupNumber) {
+    if (!CollectionUtils.isEmpty(group.getPopulations())) {
+      boolean hasMultipleIps =
+          group.getPopulations().stream()
+                  .filter(pop -> pop.getName().equals(PopulationType.INITIAL_POPULATION))
+                  .count()
+              > 1;
+
+      for (int index = 0; index < group.getPopulations().size(); index++) {
+        Population population = group.getPopulations().get(index);
+        String popDisplayId =
+            getPopulationDisplayId(population, groupNumber, hasMultipleIps, index);
+        population.setDisplayId(popDisplayId);
+      }
+    }
+  }
+
+  static void setObservationDisplayIds(Group group, String groupNumber) {
+    if (!CollectionUtils.isEmpty(group.getMeasureObservations())) {
+
+      for (int index = 0; index < group.getMeasureObservations().size(); index++) {
+        MeasureObservation observation = group.getMeasureObservations().get(index);
+        String obsDisplayId =
+            getObservationDisplayId(
+                observation, groupNumber, group.getMeasureObservations().size() > 1, index);
+        observation.setDisplayId(obsDisplayId);
+      }
+    }
+  }
+
+  static String getObservationDisplayId(
+      MeasureObservation observation, String groupNumber, boolean multipleObs, int index) {
+    String newObsDisplayId = "MeasureObservation_" + groupNumber;
+    if (multipleObs) {
+      newObsDisplayId = newObsDisplayId + "_" + (index + 1);
+    }
+    return newObsDisplayId;
+  }
+
+  static void setStratificationDisplayIds(Group group, String groupNumber) {
+    if (!CollectionUtils.isEmpty(group.getStratifications())) {
+      for (int index = 0; index < group.getStratifications().size(); index++) {
+        Stratification stratification = group.getStratifications().get(index);
+        String obsDisplayId =
+            getStratificationDisplayId(
+                stratification, groupNumber, group.getStratifications().size() > 1, index);
+        stratification.setDisplayId(obsDisplayId);
+      }
+    }
+  }
+
+  static String getStratificationDisplayId(
+      Stratification stratification, String groupNumber, boolean multipleStrats, int index) {
```

Comment:
stratification input parameter can be removed

Reply:
@RohitKandimalla removed